quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java

```diff
@@ -666,7 +665,6 @@ void configureDevMode(BuildProducer<HotDeploymentWatchedFileBuildItem> hotFiles)
 
     private Map<Spi, Map<Class<? extends Provider>, Map<String, ProviderFactory>>> loadFactories(
             Map<String, ProviderFactory> preConfiguredProviders) {
-        Config.init(new MicroProfileConfigProvider());
```

Comment:
Isn't this needed because the Config is being used by the logic below? If this isn't done at some point before this won't we only see the default which is just system properties?

Reply:
AFAIK, it should not be needed as the `initConfig()` is always executed first in the build chain before loading factories as they depend on each other. For the factories, the order of these executions should be like this: initConfig() -> configureProfile() -> setCryptoProvider() -> configureKeycloakSessionFactory().

Reply:
I've just found out it might be a little bit more problematic with the static context init during the build steps executions and bytecode recording.

I've tried an experiment with the Profile init and if the bytecode with Profile init is executed, the dependent build steps are not aware of the proceeded initialization (even with the usage of synchronization). 

I'll put a label hold on this PR and try to investigate it a little.

Reply:
Sounds good